Network/Socket.hs

```diff
@@ -100,6 +100,44 @@
 -- unexpected things would happen. There is one exception for multiple
 -- threads vs a single 'Socket': one thread reads data from a 'Socket'
 -- only and the other thread writes data to the 'Socket' only.
+--
+-- The preferred way to terminate a thread that is blocked on a call to
+-- 'Network.Socket.ByteString.recv' is to use 'shutdown'
```

Comment:
The immediate question this poses:

Why this, instead of just throwing an exception to the thread that's blocked on `recv`?

That would be the most natural to me, as that's how e.g. `async` does it.

`recv` is implemented in terms of `threadWaitRead`, so `throwTo` to it should work fine and interrupt it, shouldn't it?

Then, if you want your app to do a graceful shutdown, it can do

```hs
bracket createConnection gracefulClose $ -- ... code using recv ...
```